process/smartContract/processorV2/processV2.go

```diff
@@ -825,27 +846,50 @@ func (sc *scProcessor) doExecuteBuiltInFunction(
 	tx data.TransactionHandler,
 	acntSnd, acntDst state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
-	returnCode, vmInput, txHash, err := sc.prepareExecution(tx, acntSnd, acntDst, true)
+	failureContext := NewFailureContext(sc)
+	retCode, err := sc.doExecuteBuiltInFunctionWithoutFailureProcessing(tx, acntSnd, acntDst, failureContext)
+	if failureContext.processFail {
+		failureProcessingError := sc.processIfErrorWithAddedLogs(
+			acntSnd,
+			tx,
+			failureContext)
+		if failureProcessingError != nil {
+			err = failureProcessingError
+		}
+	}
+	return retCode, err
+}
+
+func (sc *scProcessor) doExecuteBuiltInFunctionWithoutFailureProcessing(
+	tx data.TransactionHandler,
+	acntSnd, acntDst state.UserAccountHandler,
+	failureContext *failureContext,
+) (vmcommon.ReturnCode, error) {
+	returnCode, vmInput, txHash, err := sc.prepareExecution(tx, acntSnd, acntDst, true, failureContext)
 	if err != nil || returnCode != vmcommon.Ok {
 		return returnCode, err
 	}
 
-	snapshot := sc.accounts.JournalLen()
+	failureContext.makeSnaphot()
```

Comment:
this is the second usage of these 3 functions. Suggestion to add a single function that set all these 3 values.

Reply:
In my opinion would complicate a little bit the function signatures and in one of the cases we should set the gas locked only after an error checks is made so it's possible they are not linked.